Port SWE4DS exercises (#8)
kedro-org/kedro-academy: swe4ds/environment-dependencies/README.md

# Environment and dependencies

> Your client is trying to run a very old code delivered by a former colleague, but they're unable to do so.

Reviewer: ```suggestion
Your client is trying to run very old code delivered by a former colleague, but they're unable to get it working.
```

2. At some point, you might spot a big error mentioning the `SKLEARN_ALLOW_DEPRECATED_SKLEARN_PACKAGE_INSTALL` variable. This is because the import name is **not** always equal to the package name! Refer to the top-left corner of [the PyPI page](https://pypi.org/project/scikit-learn/) for the precise command that you must run.
3. Now all the dependencies should be in place, but there's an error with the line `from sklearn.gaussian_process import GaussianProcess`. The reason is that the model name changed after this code was written. We could change the line to `from sklearn.gaussian_process import GaussianProcessRegressor`, but more code changes would be needed because of changes introduced in newer scikit-learn versions, and it's not allowed. Take a deep breath.
4. Remember that this code is old: therefore, an older version of scikit-learn will do the trick! Install `scikit-learn` 0.19 and rerun the script.

Finally! That was a bit frustrating but at least an `errorbars.png` appears in the filesystem, and the script finishes (although with lots of deprecation warnings).

### Declare the dependencies

> Create a `requirements.txt` file with the dependencies you have to install. To verify that it works:

Reviewer: Do you want to say something about where to store it (like, "in the same folder as the notebook" or "within a known location"? I know we're not assuming anything about their folder structure/project at this point, so I'm not sure, but I feel like we should offer some guidance.

Author: Indeed, will add some clarification here

Author: Done!
